src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +310,17 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool TryGetRequiredAttributeDescriptors(
+            string requiredAttributes,
+            ErrorSink errorSink,
+            out IEnumerable<TagHelperRequiredAttributeDescriptor> descriptors)
+        {
+            var parser = new RequiredAttributeParser(requiredAttributes);
```

Comment:
This is going to create a lot of useless instances and only one will ever be in use. Make `RequiredAttributeParser` stateless.